### PR TITLE
Cover PerformInteractionRequest with UT

### PR DESCRIPTION
**Pull request by indlu**
Cover PerformInteractionRequest by unit tests

### CODE REVIEW

**Comment by indlu** [comment]
@istoilovgithub , @okozlovlux, @dcherniev please, review.

File: src/components/application_manager/test/commands/mobile/perform_interaction_test.cc [hunk 1]
```diff
@@ -188,12 +170,84 @@ class PerformInteractionRequestTest
...
+    EXPECT_CALL(app_mngr_, ManageHMICommand(_))
+        .WillOnce(DoAll(SaveArg<0>(&hmi_result_msg), Return(true)));
+
+    EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _))
+        .WillOnce(DoAll(SaveArg<0>(&mobile_result_msg), Return(true)));
+  }
+
+  caller();
+
+  const hmi_apis::FunctionID::eType kHmiResult =
```
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/perform_interaction_test.cc]: @indlu , `kHmiResult` is not global
Please, check another vars

File: src/components/application_manager/test/commands/mobile/perform_interaction_test.cc [hunk 2]
```diff
@@ -354,8 +408,447 @@ TEST_F(
...
+  choise_set_1[strings::choice_set][0][strings::menu_name] = "MenuName";
+  choise_set_2[strings::choice_set][0][strings::menu_name] = "MenuName";
+
+  EXPECT_CALL(
+      *mock_application_sptr_,
+      FindChoiceSet(
+          msg_params_[strings::interaction_choice_set_id_list][0].asInt()))
+      .Times(3)
+      .WillOnce(Return(&choise_set_1))
+      // Second call will be scipped in method cycles
```
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/perform_interaction_test.cc]: @indlu , `skipped`

File: src/components/application_manager/test/commands/mobile/perform_interaction_test.cc [hunk 3]
```diff
@@ -354,8 +408,447 @@ TEST_F(
...
+
+  ::smart_objects::SmartObject choise_set_1;
+  ::smart_objects::SmartObject choise_set_2;
+  choise_set_1[strings::choice_set][0][strings::menu_name] = "MenuName";
+  choise_set_2[strings::choice_set][0][strings::menu_name] = "AnotherMenuName";
+
+  EXPECT_CALL(*mock_application_sptr_, FindChoiceSet(_))
+      .Times(6)
+      .WillOnce(Return(&choise_set_1))
+      // Second call will be scipped in method cycles
```
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/perform_interaction_test.cc]: @indlu , typo `skipped`

File: src/components/application_manager/test/commands/mobile/perform_interaction_test.cc [hunk 4]
```diff
@@ -354,8 +408,447 @@ TEST_F(
...
+  {
+    ::testing::InSequence dummy;
+    EXPECT_CALL(this->app_mngr_, ManageHMICommand(_))
+        .WillOnce(DoAll(SaveArg<0>(&first_msg), Return(true)));
+    EXPECT_CALL(this->app_mngr_, ManageHMICommand(_))
+        .WillOnce(DoAll(SaveArg<0>(&second_msg), Return(true)));
+  }
+  caller();
+
+  const hmi_apis::FunctionID::eType kFirstResult =
```
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/perform_interaction_test.cc]: @indlu , `kFirstResult` and `kSecondResult ` are not global

File: src/components/application_manager/test/commands/mobile/perform_interaction_test.cc (hunk 1 above)
Comment (indlu) [inline src/components/application_manager/test/commands/mobile/perform_interaction_test.cc]: @okozlovlux, fixed in 04b6bb3

File: src/components/application_manager/test/commands/mobile/perform_interaction_test.cc (hunk 2 above)
Comment (indlu) [inline src/components/application_manager/test/commands/mobile/perform_interaction_test.cc]: @okozlovlux, fixed in 04b6bb3

File: src/components/application_manager/test/commands/mobile/perform_interaction_test.cc (hunk 3 above)
Comment (indlu) [inline src/components/application_manager/test/commands/mobile/perform_interaction_test.cc]: @okozlovlux, fixed in 04b6bb3

File: src/components/application_manager/test/commands/mobile/perform_interaction_test.cc (hunk 4 above)
Comment (indlu) [inline src/components/application_manager/test/commands/mobile/perform_interaction_test.cc]: @okozlovlux, fixed in 04b6bb3

**Pull request merged**